### PR TITLE
Jorislive patch optional region

### PR DESCRIPTION
**Pull request by jorislive**
When your database was created in the region 'eu' or 'us' specificly the setup won't run, so I've added an optional domain parameter for the setup client.

### CODE REVIEW

File: scripts/setup.js [hunk 1]
```diff
@@ -52,7 +54,7 @@ const main = async () => {
     })
     console.log(explanation)
   }
-  let client = new faunadb.Client({ secret: adminKey })
+  let client = new faunadb.Client({ secret: adminKey, scheme: 'https', domain })
```
Comment (rts-rob) [inline scripts/setup.js:57]: Can you pull out the hardcoded 'scheme' please?

File: .env.local.example [hunk 2]
```diff
@@ -1,4 +1,5 @@
 REACT_APP_LOCAL___ADMIN=
+REACT_APP_LOCAL___ADMIN_DOMAIN=<optional 'db.eu.fauna.com' or 'db.us.fauna.com'>
```
Comment (rts-rob) [inline .env.local.example:2]: Please move these optional values to a comment around line 37 in scripts/setup.js 

This way, anyone using the classic Region Group doesn't need to set a value here.

Line should read:
```
REACT_APP_LOCAL___ADMIN_DOMAIN=
```

**Review by rts-rob — CHANGES_REQUESTED** [review]
Hey @jorislive - thank you for submitting this!

I've requested a few convention-driven changes. With those in place, we'll be g2g.

File: scripts/setup.js (hunk 1 above)
Comment (jorislive) [reply scripts/setup.js]: 👍

File: .env.local.example (hunk 2 above)
Comment (jorislive) [reply .env.local.example]: 👍

**Comment by jorislive** [comment]
Hi Rob,

I followed up on your comments and also added the domain functionality to the query manager. Furthermore I came accross a missing import statement and fixed that as well.

Cheers Joris

File: src/fauna/queries/auth.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import faunadb from 'faunadb'
+import faunadb, {Identify} from 'faunadb'
```
Comment (rts-rob) [inline src/fauna/queries/auth.js:1]: Good catch; thanks!

**Review by rts-rob — CHANGES_REQUESTED** [review]
Thanks for catching the missing import.

This PR needs modifications to src/fauna/queries/*.spec.js (auth, fweets, rate-limiting, search) to make all the tests pass with US and EU keys.

If you're willing to make those updates, that would be great. If that's too big an ask, I'll put them in. Simplest way would be to add me as a contributor to this fork, if you're okay with that. Otherwise I'll fork it and PR back into here.

**Comment by jorislive** [comment]
> Thanks for catching the missing import.
> 
> This PR needs modifications to src/fauna/queries/*.spec.js (auth, fweets, rate-limiting, search) to make all the tests pass with US and EU keys.

I've implemented the domain functionality (see last commit), but I wasn't able to test it properly, I've tried to run tests (both in a git clone with 'regional' database and a git clone with 'classic' database. Since there's no instruction about setting up the test-suite I assumed that I needed to create a .env.test.local file with the same secret as used in the .env.local file entry REACT_APP_LOCAL___ADMIN, but I wasn't able to pass all the tests. So I'm doing something wrong..

**Comment by rts-rob** [comment]
Thanks, @jorislive - I think the difference was using three underscores for the domain vs. two as defined in .env.test.local.example

I've pushed a few changes. These all work for certain definitions of "work," but the US and EU tests are intermittently failing. My hypothesis is that this is a test setup issue, where we're creating dependent resources in parallel, and the smaller Region Groups are just too fast. Great problem to have!

I'm going to have a look at serializing the resource setup. If that works, we should be g2g.

**Comment by jorislive** [comment]
@rts-rob 

> Thanks, @jorislive - I think the difference was using three underscores for the domain vs. two as defined in .env.test.local.example
> 
> I've pushed a few changes. These all work for certain definitions of "work," but the US and EU tests are intermittently failing. My hypothesis is that this is a test setup issue, where we're creating dependent resources in parallel, and the smaller Region Groups are just too fast. Great problem to have!
> 
> I'm going to have a look at serializing the resource setup. If that works, we should be g2g.

I'm not really sure if I fully understand that 'too fast' could be problematic, but it does sound like a luxury problem indeed!
Good luck with serializing the resource setup! 

It was nice for me to have a look at the code, it provided me with good insights in how to create a more advanced setup with Fauna. For me Fauna was already the weapon of choice for the client of my new project, so I needed to deep dive into it. But I must say I'm starting to become a real fan of it's abillities!

**Comment by rts-rob** [comment]
Hello again @jorislive - we're going to close this PR in favor of #79 (just merged).

79 takes the approach of recommending users run this in the Classic/Global Region Group. This is a simple documentation fix, and more appropriate for a learning project.

We explored making the fixes to enable Fwitter to run in all Region Groups, but it requires a significant refactor to get the tests to pass. This refactor would be moving schema generation onto [Fauna Schema Migrate](https://github.com/fauna-labs/fauna-schema-migrate) to avoid resource creation contention issues in the smaller, faster Region Groups.

We're currently working on a new, opinionated flagship application that incorporates Region Group support from the beginning. This is expected to be released by end of October 2021, with additional features layered in throughout the coming year. The level of effort required to move Fwitter to FSM is too high when that work will ultimately be replaced in less than a quarter.

Thanks again for submitting this PR!

**Pull request closed without merging**